zerver/models.py

```diff
@@ -3069,19 +3069,33 @@ def is_status_message(content: str, rendered_content: str) -> bool:
     class Meta:
         indexes = [
             GinIndex("search_tsvector", fastupdate=False, name="zerver_message_search_tsvector"),
-            models.Index(Upper("subject"), name="upper_subject_idx"),
-            models.Index("date_sent", name="zerver_message_date_sent_3b5b05d8"),
             models.Index(
-                "recipient",
+                "realm_id", "recipient_id", "id", name="zerver_message_realm_recipient_id"
+            ),
+            models.Index(
+                "realm_id",
+                "sender_id",
+                "recipient_id",
+                name="zerver_message_realm_sender_recipient",
+            ),
+            models.Index(
+                "realm_id", "sending_client_id", name="zerver_message_realm_sending_client"
+            ),
+            models.Index("realm_id", Upper("subject"), name="zerver_message_realm_upper_subject"),
+            models.Index("realm_id", "date_sent", name="zerver_message_realm_date_sent"),
```

Comment:
Do you remember what this index is used for? I had been thinking it was related to `update_first_visible_message_id`, but checked and that just needs a `realm_id, id` index.

Reply:
Analytics.